src/main/java/com/mattworzala/canary/internal/util/ui/itembehavior/GenericItemBehavior.java

```diff
@@ -0,0 +1,55 @@
+package com.mattworzala.canary.internal.util.ui.itembehavior;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.entity.Player;
+
+public class GenericItemBehavior implements ItemBehavior {
+    private String baseCommand;
+    private CommandClick leftClick;
+    private CommandClick rightClick;
+
+    public GenericItemBehavior(String baseCommand, CommandClick leftClick, CommandClick rightClick) {
+        this.baseCommand = baseCommand;
+        this.leftClick = leftClick;
+        this.rightClick = rightClick;
+    }
+
+    @Override
+    public void onLeftClick(Player player, Point point) {
+        if (leftClick != null) {
+            try {
+                leftClick.handle(player, point)
+                        .thenAccept(str -> {
+                            if (str != null) {
```

Comment:
Could just use exceptions here instead of null checks perhaps? Since this should only happen on misconfig.